dev_setup.sh

```diff
@@ -87,23 +63,160 @@ done
+            autoupdate=false
+            break
+            ;;
+        esac
+    done
+
+    #  Pull down mimic source?  Most will be happy with just the package
+    if [[ ${opt_forcemimicbuild} == false && ${opt_skipmimicbuild} == false ]] ; then
+        sleep 0.5
+        echo
+        echo "Mycroft uses its Mimic technology to speak to you.  Mimic can run both"
+        echo "locally and from a server.  The local Mimic is more robotic, but always"
+        echo "available regardless of network connectivity.  It will act as a fallback"
+        echo "if unable to contact the Mimic server."
+        echo
+        echo "However, building the local Mimic is time consuming -- it can take hours"
+        echo "on slower machines.  This can be skipped, but Mycroft will be unable to"
+        echo "talk if you lose network connectivity.  Would you like to build Mimic"
+        echo "locally?"
+        echo -e -n "Choice [${CYAN}Y${RESET}/${CYAN}N${RESET}]: "
+        while true; do
+            read -N1 -s key
+            case $key in
+            [Yy])
+                echo -e "${HIGHLIGHT} $key - Mimic will be built ${RESET}"
+                break
+                ;;
+            [Nn])
+                echo -e "${HIGHLIGHT} $key - skip Mimic build ${RESET}"
+                opt_skipmimicbuild=true
+                break
+                ;;
+            esac
+        done
+    fi
+
+    # Create a link to the 'skills' folder.
+    sleep 0.5
+    echo
+    echo "The standard location for Mycroft skills is under /opt/mycroft/skills."
+
+    if [[ ! -d /opt/mycroft/skills ]] ; then
+        echo "This script will create that folder for you.  This requires sudo"
+        echo "permission and might ask you for a password..."
+        sudo mkdir -p /opt/mycroft/skills
+        sudo chown -R $USER:$USER /opt/mycroft
+        echo "Created!"
+    fi
+    if [[ ! -d skills ]] ; then
+        ln -s /opt/mycroft/skills skills
+        echo "For convenience, a soft link has been created called 'skills' which leads"
+        echo "to /opt/mycroft/skills."
+    fi
+
+    # Add mycroft-core/bin to the .bashrc PATH?
+    sleep 0.5
+    echo
+    echo "There are several Mycroft helper commands in the bin folder.  These"
+    echo "can be added to your system PATH, making it simpler to use Mycroft."
+    echo "Would you like this to be added to your PATH in the .profile?"
+    echo -e -n "Choice [${CYAN}Y${RESET}/${CYAN}N${RESET}]: "
+    while true; do
+        read -N1 -s key
+        case $key in
+        [Yy])
+            echo -e "${HIGHLIGHT} $key - Adding mycroft-core/bin to your PATH ${RESET}"
+            echo '' >> ~/.profile
+            echo '# set path so it includes Mycroft utilities' >> ~/.profile
+            echo 'if [ -d "'${TOP}'/bin" ] ; then' >> ~/.profile
+            echo '    PATH="'${TOP}'/bin:$PATH"' >> ~/.profile
+            echo 'fi' >> ~/.profile
+            echo -e "Type ${CYAN}mycroft-help${RESET} to see available commands."
```

Comment:
Should we prepare a `.mycroft_profile` file that's sourced in the .profile instead? Would probably be better in case of future additions and if a new mycroft installation is done the `.mycroft_profile` file can just be overwritten.

I don't think we should place the mycroft-core/bin folder as the first lookup folder.

Reply:
Good idea.  I implemented this, along with a comment in the .mycroft_profile indicating that the file should not be customized as it might be replaced.